### PR TITLE
Static config for syncs

### PR DESCRIPTION
**Pull request by wmathurin**
Two new methods on SmartSyncSDKManager:
* `setupGlobalSyncsFromDefaultConfig`
Call at any time to create syncs defined in resource file`globalsyncs.json` into the default global smartstore.

* `setupUserSyncsFromDefaultConfig`
Call once a user has logged in to create syncs defined in resource file `usersyncs.json` into the user's default smartstore.

Notes:
* The config files should be at the root of the resource bundle for native apps and under www/ in hybrid apps.
* The JSON format is based on what we use in our Cordova plugin and React Native module.
* It just does a bunch of `createSyncDown`  and `createSyncUp` calls, it will skip syncs if they already exist.
* The only feedback is through the logs - that's intentional, those config files are static (can't be changed at runtime) so the developer should get them right while developing the app.

The sample application SmartSyncExplorer now defines its syncs in a config file.
I also added a new test suite called SFSDKSyncsConfigTests.

Example of JSON config file
```json
{
  "syncs": [
    {
      "syncName": "syncDownContacts",
      "syncType": "syncDown",
      "soupName": "contacts",
      "target": {"type":"soql", "query":"SELECT FirstName, LastName, Title, MobilePhone, Email, Department, HomePhone FROM Contact LIMIT 10000"},
      "options": {"mergeMode":"OVERWRITE"}
    },
    {
      "syncName": "syncUpContacts",
      "syncType": "syncUp",
      "soupName": "contacts",
      "target": {"createFieldlist":["FirstName", "LastName", "Title", "MobilePhone", "Email", "Department", "HomePhone"]},
      "options": {"fieldlist":["Id", "FirstName", "LastName", "Title", "MobilePhone", "Email", "Department", "HomePhone"], "mergeMode":"LEAVE_IF_CHANGED"}
    }
  ]
}
```

### CODE REVIEW

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewConfig.m [hunk 1]
```diff
@@ -25,18 +25,12 @@
 
 #import "SFHybridViewConfig.h"
 #import <SalesforceAnalytics/SFSDKLogger.h>
-#import <SalesforceSDKCore/SFJsonUtils.h>
 #import <SalesforceSDKCore/SFSDKResourceUtils.h>
 #import "SFSDKHybridLogger.h"
+#import <SalesforceSDKCore/SFSDKResourceUtils.h>
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewConfig.m]: Looks like a duplicate import. It has been imported 2 lines above.

File: libs/SmartSync/SmartSyncTests/SFSDKSyncsConfigTests.m [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+@end
+
+@implementation SFSDKSyncsConfigTests
+
+#pragma mark - setup and teardown
+
+- (void) setUp
+{
+    [super setUp];
+    [SFSDKSmartSyncLogger setLogLevel:DDLogLevelDebug];
```
Comment (bhariharan) [inline libs/SmartSync/SmartSyncTests/SFSDKSyncsConfigTests.m]: I don't know if we need this. It should be `Debug` by default in a debug build.

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR, try to keep changes smaller if you can</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

### Tests: 

Classname | Name|
--- | ---|
MetadataManagerTests | testCommonMRUObjectsFromServer|
MetadataManagerTests | testCommonMRUObjectsFromServer|
SyncManagerTests | testCleanResyncGhostsForMRUTarget|
SyncManagerTests | testCleanResyncGhostsForMRUTarget|
SyncManagerTests | testCleanResyncGhostsForMRUTarget|


## Current coverage for SalesforceAnalytics.framework is `66.64%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceHybridSDK.framework is `0.00%`
Files changed | - | - 
--- | --- | ---
SalesforceHybridSDKManager.m | `0.00%` | :skull:
SFHybridViewConfig.m | `0.00%` | :skull:
SFHybridViewController.m | `0.00%` | :skull:
SalesforceHybridSDKManager.m | `0.00%` | :skull:
SFHybridViewController.m | `0.00%` | :skull:
SFHybridViewConfig.m | `0.00%` | :skull:

---
## Current coverage for SalesforceSDKCore.framework is `50.78%`
Files changed | - | - 
--- | --- | ---
SFSDKResourceUtils.m | `58.00%` | :warning:
SFSDKResourceUtils.m | `58.00%` | :warning:

---
## Current coverage for SmartStore.framework is `62.81%`
Files changed | - | - 
--- | --- | ---
SmartStoreSDKManager.m | `41.94%` | :no_entry_sign:
SmartStoreSDKManager.m | `41.94%` | :no_entry_sign:
SFSDKStoreConfig.m | `75.00%` | :warning:
SFSDKStoreConfig.m | `75.00%` | :warning:

---
## Current coverage for SmartSync.framework is `78.88%`
Files changed | - | - 
--- | --- | ---
SFSDKSyncsConfig.m | `85.00%` | :white_check_mark:
SFSDKSyncsConfig.m | `85.00%` | :white_check_mark:
SFSmartSyncSyncManager.m | `85.29%` | :white_check_mark:
SFSmartSyncSyncManager.m | `85.29%` | :white_check_mark:
SFSyncState.m | `94.29%` | :white_check_mark:
SFSyncState.m | `94.29%` | :white_check_mark:
SmartSyncSDKManager.m | `100.00%` | :white_check_mark:
SmartSyncSDKManager.m | `100.00%` | :white_check_mark:

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_CI-result">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by wmathurin** [comment]
@brandonpage the salesforcehybridsdk coverage [here](https://forcedotcom.github.io/SalesforceMobileSDK-TestResults/iOS-PR-results/788/xcov_output/index.html) is off.

**Pull request merged**